### ISSUE TITLE
Version file missing from user-directory config

### ISSUE THREAD

**Issue by zycon** [kubernetes-sigs/cluster-api#3702]
**What steps did you take and what happened:**
clusterctl version -v=10

**What did you expect to happen:**

display version or something, have not seen before

**Anything else you would like to add:**
Error: unable to verify clusterctl version: unable to write version state file: open /Users/zycon/.cluster-api/version.yaml: no such file or directory

clusterctl version: &version.Info{Major:"0", Minor:"3", GitVersion:"v0.3.9", GitCommit:"e1f67d8ceb1d5b30ef967035f7a0d1b1ee088b37", GitTreeState:"clean", BuildDate:"2020-09-01T02:45:38Z", GoVersion:"go1.13.14", Compiler:"gc", Platform:"darwin/amd64"}
No default config file available
Error: unable to verify clusterctl version: unable to write version state file: open /Users/tibin/.cluster-api/version.yaml: no such file or directory
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.glob..func9
	/workspace/cmd/clusterctl/cmd/root.go:58
github.com/spf13/cobra.(*Command).execute
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:857
github.com/spf13/cobra.(*Command).ExecuteC
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950
github.com/spf13/cobra.(*Command).Execute
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.Execute
	/workspace/cmd/clusterctl/cmd/root.go:69
main.main
	/workspace/cmd/clusterctl/main.go:25
runtime.main
	/usr/local/go/src/runtime/proc.go:203
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1357

**Environment:**

- Cluster-api version: 
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`): version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.6-beta.0", GitCommit:"e7f962ba86f4ce7033828210ca3556393c377bcc", GitTreeState:"clean", BuildDate:"2020-01-15T08:26:26Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"darwin/amd64"}
- OS (e.g. from `/etc/os-release`): mac

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

**Comment by vincepri** [comment]
/milestone Next

Please triage when possible

**Comment by wfernandes** [comment]
@zycon I wasn't able to reproduce this behavior.
Steps:
1. I pulled down the `clusterctl-darwin-amd64` binary from our releases page for release v0.3.9 and ensured it was executable on my machine.
1. I removed the default directory path `~/.cluster-api/`
1. Ran the command
    ```shell
    $ ./clusterctl-darwin-amd64 version -v=10
    clusterctl version: &version.Info{Major:"0", Minor:"3", GitVersion:"v0.3.9", 
    GitCommit:"e1f67d8ceb1d5b30ef967035f7a0d1b1ee088b37", GitTreeState:"clean", BuildDate:"2020-09-01T02:45:38Z", GoVersion:"go1.13.14", Compiler:"gc", Platform:"darwin/amd64"}
    No default config file available
    ```
1. Version file was created.
    ```shell
    $ cat ~/.cluster-api/version.yaml
    LastCheck: "2020-09-29T13:51:32.811924-06:00"
    LatestRelease:
      URL: https://github.com/kubernetes-sigs/cluster-api/releases/tag/v0.3.9
      Version: v0.3.9
    ```

Can you please verify the shasum of the clusterctl binary you are using?
```shell
$ shasum -a 256 clusterctl-darwin-amd64
5d5accbafa85da99e2f3d45188a4bce4531cdeacbf5bba04f1352c9dcbad9eac  clusterctl-darwin-amd64
```

**Comment by shysank** [comment]
@wfernandes @zycon 
I was able to reproduce it in `v0.3.9`.

But, this has been fixed in `0.3.10` [here](https://github.com/kubernetes-sigs/cluster-api/commit/a9c75fb2374d5cdd5e9e7294a5f404bd2d92f0ee). I was able to verify this.

```
±  |tag:v0.3.9 ✓| → rm -rf ~/.cluster-api

 2020-10-06 12:32:45 ⌚ |ruby-2.6.3| sankarans-a01 in ~/workspace/cluster-api
±  |tag:v0.3.9 ✓| → make clusterctl
go build -ldflags "-X 'sigs.k8s.io/cluster-api/cmd/version.buildDate=2020-10-06T19:32:49Z' -X 'sigs.k8s.io/cluster-api/cmd/version.gitCommit=e1f67d8ceb1d5b30ef967035f7a0d1b1ee088b37' -X 'sigs.k8s.io/cluster-api/cmd/version.gitTreeState=clean' -X 'sigs.k8s.io/cluster-api/cmd/version.gitMajor=0' -X 'sigs.k8s.io/cluster-api/cmd/version.gitMinor=3' -X 'sigs.k8s.io/cluster-api/cmd/version.gitVersion=v0.3.9' -X 'sigs.k8s.io/cluster-api/cmd/version.gitReleaseCommit=e1f67d8ceb1d5b30ef967035f7a0d1b1ee088b37'" -o bin/clusterctl sigs.k8s.io/cluster-api/cmd/clusterctl

 2020-10-06 12:32:52 ⌚ |ruby-2.6.3| sankarans-a01 in ~/workspace/cluster-api
±  |tag:v0.3.9 ✓| → ./bin/clusterctl version
clusterctl version: &version.Info{Major:"0", Minor:"3", GitVersion:"v0.3.9", GitCommit:"e1f67d8ceb1d5b30ef967035f7a0d1b1ee088b37", GitTreeState:"clean", BuildDate:"2020-10-06T19:32:49Z", GoVersion:"go1.14.1", Compiler:"gc", Platform:"darwin/amd64"}
Error: unable to verify clusterctl version: unable to write version state file: open /Users/shyamsankaran/.cluster-api/version.yaml: no such file or directory

 2020-10-06 12:32:57 ⌚ |ruby-2.6.3| sankarans-a01 in ~/workspace/cluster-api
±  |tag:v0.3.9 ✓| → git checkout tags/v0.3.10
Previous HEAD position was e1f67d8ce Merge pull request #3562 from vincepri/disallow1190upgrades
HEAD is now at af6630920 Merge pull request #3685 from richardcase/eks-controlplane

 2020-10-06 12:33:21 ⌚ |ruby-2.6.3| sankarans-a01 in ~/workspace/cluster-api
±  |tag:v0.3.10 ✓| → make clusterctl
go build -ldflags "-X 'sigs.k8s.io/cluster-api/cmd/version.buildDate=2020-10-06T19:33:27Z' -X 'sigs.k8s.io/cluster-api/cmd/version.gitCommit=af6630920560ca0e12179897b96d6ea8bd830b63' -X 'sigs.k8s.io/cluster-api/cmd/version.gitTreeState=clean' -X 'sigs.k8s.io/cluster-api/cmd/version.gitMajor=0' -X 'sigs.k8s.io/cluster-api/cmd/version.gitMinor=3' -X 'sigs.k8s.io/cluster-api/cmd/version.gitVersion=v0.3.10' -X 'sigs.k8s.io/cluster-api/cmd/version.gitReleaseCommit=af6630920560ca0e12179897b96d6ea8bd830b63'" -o bin/clusterctl sigs.k8s.io/cluster-api/cmd/clusterctl

 2020-10-06 12:33:31 ⌚ |ruby-2.6.3| sankarans-a01 in ~/workspace/cluster-api
±  |tag:v0.3.10 ✓| → ./bin/clusterctl version
clusterctl version: &version.Info{Major:"0", Minor:"3", GitVersion:"v0.3.10", GitCommit:"af6630920560ca0e12179897b96d6ea8bd830b63", GitTreeState:"clean", BuildDate:"2020-10-06T19:33:27Z", GoVersion:"go1.14.1", Compiler:"gc", Platform:"darwin/amd64"}

```

**Comment by wfernandes** [comment]
Ahh..ok. Yeah I believe that bug was fixed for the initial creation of the directory (`~/.cluster-api`) if it didn't exist. 
Since this is fixed, I'm going to go ahead and close this. Thanks.
/close
